medallion/views/__init__.py

```diff
@@ -43,3 +43,35 @@ def decorated_function(*args, **kwargs):
         return f(*args, **kwargs)
 
     return decorated_function
+
+
+def admin_only_endpoint(f):
+    """Decorator for endpoint functions to allow access to admin users only."""
+    @wraps(f)
+    def decorated_function(*args, **kwargs):
+        if not AuthService.is_user_admin(auth.current_user()):
+            return Response(
+                response="Unauthorized Access",
+                status=401,
```

Comment:
this error should be 403 forbidden, 401 is meant to be used when the user is not supplying any authn means